gravitee-apim-console-webui/src/management/api/policy-studio/debug/components/policy-studio-debug-timeline-hover/policy-studio-debug-timeline-hover.directive.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, ElementRef, HostListener, Input, OnDestroy, OnInit, Renderer2 } from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { PolicyStudioDebugTimelineHoverService } from './policy-studio-debug-timeline-hover.service';
+
+@Directive({
+  selector: '[timelineHover]',
+})
+export class PolicyStudioDebugTimelineHoverComponent implements OnInit, OnDestroy {
```

Comment:
This directive and its service might not be specific to the Timeline and work for any element. WDYT?

Reply:
Yes it could, that's why I made the directive and not used the service in the components. But I didn't put it in shared because I don't see the other needs. and I was more afraid that we have specific timeline cases in the future
So I don't know if "ca vaut le coup " to putting it in shared now 🤷‍♂️ but why not - i'm 50/50

Reply:
👍 Yes, it's "vaut plus le coup" until we see the evolution